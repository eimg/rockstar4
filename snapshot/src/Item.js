import React from 'react';

const Item = props => (
    <li>
        {props.name} (${props.price})
    </li>
);

export default Item;
