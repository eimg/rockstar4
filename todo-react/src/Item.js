import React from "react";

class Item extends React.Component {
    render() {
        return (
            <li>
                {this.props.item.status
                    ? <input type="checkbox" checked
                        onChange={this.props.undo(this.props.item._id)} />
                    : <input type="checkbox"
                        onChange={this.props.done(this.props.item._id)} />
                }

                {this.props.item.subject}
                <a href="/#" onClick={this.props.remove(this.props.item._id)}>
                    &times;
                </a>
            </li>
        )
    }
}

export default Item;
