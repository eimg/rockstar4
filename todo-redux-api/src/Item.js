import React from "react";

const Item = props => (
    <li>
        {props.item.status
            ? <input type="checkbox" checked
                onChange={() => props.undo(props.item._id)} />
            : <input type="checkbox"
                onChange={() => props.done(props.item._id)} />
        }

        {props.item.subject}
        <a href="/#" onClick={() => props.remove(props.item._id)}>
            &times;
        </a>
    </li>
)

export default Item;
