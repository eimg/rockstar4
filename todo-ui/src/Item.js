import React from "react";

import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import IconButton from "@material-ui/core/IconButton";
import Checkbox from "@material-ui/core/Checkbox";
import DeleteIcon from "@material-ui/icons/Delete";

class Item extends React.Component {
    render() {
        return (
            <ListItem>
                {this.props.item.status
                    ? <Checkbox checked="true"
                        onChange={this.props.undo(this.props.item._id)} />
                    : <Checkbox
                        onChange={this.props.done(this.props.item._id)} />
                }

                <ListItemText primary={this.props.item.subject} />

                <IconButton onClick={this.props.remove(this.props.item._id)}>
                    <DeleteIcon />
                </IconButton>
            </ListItem>
        )
    }
}

export default Item;
