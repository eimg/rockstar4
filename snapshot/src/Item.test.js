import React from 'react';
import Item from './Item.js';
import renderer from 'react-test-renderer';

test('Item Component', () => {
    const tree = renderer.create(
        <Item name="Apple" price="2.99" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
