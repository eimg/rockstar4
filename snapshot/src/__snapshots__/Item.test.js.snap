// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component 1`] = `
<li>
  Apple
   ($
  2.99
  )
</li>
`;
